src/traversing.js

```diff
@@ -168,10 +159,18 @@ jQuery.each({
 		return elem.contentDocument || jQuery.merge( [], elem.childNodes );
 	}
 }, function( name, fn ) {
+	var // Sorting (which implicitly de-dupes) is required for non-adjacent traversals
+		sort = name !== "prev" && name !== "next",
```

Comment:
Using these methods on a set of more than one is not uncommon.

As for large sets, I'm not sure how common it is, but cases like selecting all cells in the first column of a table and doing something, then selecting all cells in the next column via `.next()` seems sane (from a user's perspective) and could result in a large set.

The real question, though, is what performance difference are we talking about, and for what percentage of users?